src/redux/stores/orders/ordersApi.ts

```diff
@@ -61,35 +48,40 @@ export async function approveNftToken(
     provider.getSigner(),
   );
   return contract.approve(
-    SwapERC20.getAddress(provider.network.chainId),
+    Swap.getAddress(provider.network.chainId),
     tokenId,
   );
 }
 
 export async function takeOrder(
-  order: OrderERC20 | FullOrderERC20,
+  order: FullOrder,
+  senderWallet: string,
   provider: ethers.providers.Web3Provider,
-): Promise<Transaction | AppError> {
-  return new Promise<Transaction | AppError>((resolve) => {
-    swap(provider.network.chainId, provider, order)
-      .then(tx => resolve(tx))
-      .catch((error: any) => {
-        transformUnknownErrorToAppError(error);
-      });
-  });
+): Promise<ContractTransaction | AppError> {
+  try {
+    const { chainId } = provider.network;
+    const result = await Swap.getContract(provider.getSigner(), chainId).swap(
+      senderWallet,
+      '0',
+      order,
+    );
+
+    return result;
+  } catch (error: any) {
+    console.error(error);
+    return transformUnknownErrorToAppError(error);
+  }
 }
 
-export async function check(
-  order: OrderERC20,
+export async function checkOrder(
+  order: FullOrder,
   senderWallet: string,
-  chainId: number,
-  signer?: ethers.providers.JsonRpcSigner,
+  provider: ethers.providers.Web3Provider,
 ): Promise<AppError[]> {
-  const errors = (await new SwapERC20(chainId, signer).check(
-    order,
-    senderWallet,
-    signer,
-  )) as SwapError[];
+  const { chainId } = provider.network;
+
+  const result = await Swap.getContract(provider.getSigner(), chainId).check(senderWallet, order);
+  const errors = checkResultToErrors(result[1], result[0]) as SwapError[];
```

Comment:
maybe check the result lenght before accessing indexes ?

Reply:
Yes good point.